server/src/server.ts

```diff
@@ -334,82 +342,24 @@ CallStack (from HasCallStack):
-				if (i < (tokens.length - 1)) {
-					actualMessage += ":"; // add back the colons in between
-				}
-			}
+		const linePos = errorJson.ce_position[0] - 1;
+		const charPos = errorJson.ce_position[1] - 1;
+		const suggestions = errorJson.ce_suggestions;
+		const actualMessage = errorJson.ce_errorMessage;
 
-			start = { line: 0, character: 0 }; // generic error highlights everything
-			end = { line: 9999, character: 9999 };
-		} else {
-			for (var i = 4; i < tokens.length; i++) { // start after line/pos
-				actualMessage += tokens[i];
-				if (i < (tokens.length - 1)) {
-					actualMessage += ":"; // add back the colons in between
-				}
-			}
-
-			start = { line: linePos - 1, character: charPos - 1 } // Reach compiler numbers starts at 1
-
-			// Get list of suggestions from compiler
-			const SUGGESTIONS_PREFIX = "Did you mean: [";
-			const SUGGESTIONS_SUFFIX = "]";
-			var indexOfSuggestions = actualMessage.indexOf(SUGGESTIONS_PREFIX);
-			var suggestions;
-			if (indexOfSuggestions != -1) {
-				suggestions = actualMessage.substring(indexOfSuggestions + SUGGESTIONS_PREFIX.length, actualMessage.lastIndexOf(SUGGESTIONS_SUFFIX));
-			}
-			connection.console.log(`Parsed suggestions: ${suggestions}`);
-
-			// Get the problematic string (before the list of suggestions)
-			if (suggestions !== undefined) {
-				var problematicString;
-				var messageWithoutSuggestions = actualMessage.substring(0, indexOfSuggestions);
-				let messageWithoutSuggestionsTokens: string[] = messageWithoutSuggestions.split(" ");
-				//connection.console.log(`messageWithoutSuggestionsTokens: ${messageWithoutSuggestionsTokens}`);
-
-				if (tokens[4] == "dot") {
-					// e.g.: error, ./.index.rsh.temp,8,8,dot, AApp is not a field of Reach. Did you mean, ["App"]
-					problematicString = messageWithoutSuggestionsTokens[1];
-					connection.console.log(`Problem found in string: ${problematicString}`);
-
-					start = { line: linePos - 1, character: charPos } // Reach compiler numbers starts at 1, but skip the dot in highlighting
-					end = { line: linePos - 1, character: charPos + problematicString.length };
-				} else {
-					problematicString = messageWithoutSuggestionsTokens[messageWithoutSuggestionsTokens.length - 2]; // last space is a token too
-					problematicString = problematicString.substring(0, problematicString.length - 1); // remove trailing period at end of sentence 
-					connection.console.log(`Problem found in string: ${problematicString}`);
-
-					end = { line: linePos - 1, character: charPos - 1 + problematicString.length };
-				}
-			} else {
-				end = { line: linePos, character: 0 } // until end of line, or equivalently the next line
-			}
-		}
+		const start = { line: linePos, character:charPos };
 
 		let location: ErrorLocation = {
-			range: {
-				start: start,
-				end: end
-			},
+			// Reachc does not output position range, so this hack will just squiggle the word token.
+			range: { start: start, end: start },
```

Comment:
This diagnostic is used in the getCodeActions() function below, so the `end` of the range needs to be the actual end of the word token.  Otherwise if you hover over the error, choose Quick Fix and select a replacement, the replacement is inserted before the word instead of actually replacing it.  If Reach does not provide the end position, perhaps it should provide the word token itself so the length can be calculated here.  Alternatively, parse the word token from the error message (but that is ugly and defeats the point of using JSON, but it could be temporary workaround until the end position is available in the JSON).

Reply:
I've updated the Reach JSON error message with a `ce_offendingToken` field. If the offending token property exists on an error, it will use the word length or default to the end of the line. Tested the quick fix with identifiers and object properties, and it's working as expected now